### PR TITLE
Remove hard coded census date

### PR DESCRIPTION
**Pull request by lauraedwards1**
### PR Context
The census date has been updated to be 21/03/2021 for all schemas, however some instances remain where the census date has been hard coded to be 13/10/2019. We found and replaced hard coded dates with census date variable. In order to use the census date in contents lists, we had to add extra changes to validator and eq-translations.

### CODE REVIEW

**Comment by petechd** [comment]
[Remove hard-coded census dates from schemas (S)](https://trello.com/c/d8k4s5QA/3676-remove-hard-coded-census-dates-from-schemas-s)

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
     runs-on: ubuntu-latest
     services:
       validator:
-        image: onsdigital/eq-questionnaire-validator:latest
+        image: onsdigital/eq-questionnaire-validator:remove-hard-coded-census-date
```
Comment (JamesGardiner) [inline .github/workflows/pull_request.yml]: revert before merge

File: Pipfile [hunk 2]
```diff
@@ -7,7 +7,7 @@ verify_ssl = true
 coloredlogs = "*"
 
 [packages]
-eq-translations = {editable = true,git = "https://github.com/ONSDigital/eq-translations.git",ref = "master"}
+eq-translations = {editable = true,git = "https://github.com/ONSDigital/eq-translations.git",ref = "remove-hard-coded-census-date"}
```
Comment (JamesGardiner) [inline Pipfile]: revert before merge

File: scripts/run_validator.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
 
-branch=latest
+branch=remove-hard-coded-census-date
```
Comment (JamesGardiner) [inline scripts/run_validator.sh]: revert before merge

File: source/jsonnet/england-wales/ccs/blocks/individual/past_usual_household_address.jsonnet [hunk 4]
```diff
@@ -33,9 +33,9 @@ local question(title) = {
   ],
 };
 
-local nonProxyTitle = 'One year ago, on 13 October 2018, what was your usual address?';
+local nonProxyTitle = 'One year ago, on 21 March 2020, what was your usual address?';
 local proxyTitle = {
-  text: 'One year ago, on 13 October 2018, what was {person_name_possessive} usual address?',
+  text: 'One year ago, on 21 March 2020, what was {person_name_possessive} usual address?',
```
Comment (JamesGardiner) [inline source/jsonnet/england-wales/ccs/blocks/individual/past_usual_household_address.jsonnet]: These dates should be calculated. If it's not much work ideally as part of this ticket. If it's a fair bit more involved, it's worth creating another ticket to update them.

Comment (petechd) [reply source/jsonnet/england-wales/ccs/blocks/individual/past_usual_household_address.jsonnet]: I think it means adding new transform to runner and transform rules-definitions to validator, then updating eq-schemas. I would put it on a separate card.

File: source/jsonnet/england-wales/ccs/blocks/accommodation/accommodation_introduction.jsonnet [hunk 5]
```diff
@@ -1,11 +1,17 @@
+local placeholders = import '../../../lib/placeholders.libsonnet';
 {
```
[the remark this answers is not in the document]
Comment (lauraedwards1) [reply source/jsonnet/england-wales/ccs/blocks/accommodation/accommodation_introduction.jsonnet]: 👍

File: source/jsonnet/england-wales/ccs/blocks/individual/another_uk_address.jsonnet [hunk 6]
```diff
@@ -1,14 +1,28 @@
 local placeholders = import '../../../lib/placeholders.libsonnet';
 local rules = import 'rules.libsonnet';
 
-local nonProxyTitle = 'Is there another UK address where you may have been included on a census questionnaire because you were a usual resident, or staying overnight there on Sunday 13 October 2019?';
+local nonProxyTitle = {
+  text: 'Is there another UK address where you may have been included on a census questionnaire because you were a usual resident, or staying overnight there on Sunday {census_date}?',
```
Comment (ajmaddaford) [inline source/jsonnet/england-wales/ccs/blocks/individual/another_uk_address.jsonnet]: My initial reaction was that the day should be handled by the `format_date` placeholder transform (the day can be output by adding `EEEE` to the `date_to_format` property). Unfortunately the Welsh translation via Babel is then incorrect for our usage. It translates "Sunday" as "Dydd Sul", whereas our current translations use "ddydd Sul" (ddydd is lowercase mutates due to the "on" before the day).

So, in summary, I think this usage is acceptable.

File: source/jsonnet/england-wales/ccs/blocks/individual/another_uk_address.jsonnet [hunk 7]
```diff
@@ -24,7 +38,7 @@ local question(title) = {
         {
           label: 'Previous home',
           value: 'Previous home',
-          description: 'Moved since 13 October 2019',
+          description: description,
```
Comment (ajmaddaford) [inline source/jsonnet/england-wales/ccs/blocks/individual/another_uk_address.jsonnet]: I'm not sure why we keep creating variables for small blocks that aren't being re-used anywhere. Here it would be clearer if the description Jsonnet was inline.

File: source/jsonnet/england-wales/ccs/blocks/individual/employment_status.jsonnet [hunk 8]
```diff
@@ -56,11 +56,18 @@ local question(title) = {
...
+local nonProxyTitle = {
+  text: 'During the week of 15 to {census_date}, were you doing any of the following?',
+  placeholders: [
+    placeholders.censusDate,
+  ],
+};
+
 local proxyTitle = {
-  text: 'During the week of 7 to 13 October 2019, was {person_name} doing any of the following?',
+  text: 'During the week of 15 to {census_date}, was {person_name} doing any of the following?',
```
Comment (ajmaddaford) [inline source/jsonnet/england-wales/ccs/blocks/individual/employment_status.jsonnet]: We need to be aware that this may cause some problems with translations. Without knowing the format of `census_date`, the translator may not understand.

File: source/jsonnet/england-wales/ccs/blocks/individual/past_usual_household_address.jsonnet (hunk 4 above)
Comment (ajmaddaford) [reply source/jsonnet/england-wales/ccs/blocks/individual/past_usual_household_address.jsonnet]: We could consider making it a placeholder for now and hard coding the date in the placeholder. That way we will have removed hardcoded dates from all strings.

File: source/jsonnet/northern-ireland/household/blocks/who-lives-here/who_lives_here_interstitial.jsonnet [hunk 9]
```diff
@@ -21,8 +27,7 @@ local contentDescription = {
         title: 'You will need to know',
         list: [
           'Names of the people living at this address including anyone temporarily away or who has been or intends to be in the UK for 3 months or more.',
-          'Names of visitors staying overnight at this address on 13 October 2019',
-          // This date is currently hard coded as validator currently does not allow placeholders within list object.
+          optionDescription,
```
Comment (ajmaddaford) [inline source/jsonnet/northern-ireland/household/blocks/who-lives-here/who_lives_here_interstitial.jsonnet]: The variable name is misleading. We should consider not using a variable and inlining the placeholder Jsonnet here.

File: scripts/run_validator.sh (hunk 3 above)
Comment (petechd) [reply scripts/run_validator.sh]: Reverted the change since validator PR is merged into master now.

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (petechd) [reply .github/workflows/pull_request.yml]: This has moved after [Update to use eq-questionnaire-validator repo](https://github.com/ONSdigital/eq-questionnaire-schemas/pull/4) has merged.

Comment (petechd) [reply .github/workflows/pull_request.yml]: It has changed since this [PR](https://github.com/ONSdigital/eq-questionnaire-schemas/pull/20) got merged. Not used anymore.

File: Pipfile (hunk 2 above)
Comment (petechd) [reply Pipfile]: Done!👍

**Comment by petechd** [comment]
Revert branch change before merge
